plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/ConnectionsToStructCommand.java

```diff
@@ -0,0 +1,309 @@
+//		sourceType = (FBType)sourceType.getTypeEntry().getType();
+		AbstractLiveSearchContext.save(destinationType.getTypeEntry());
+//		destinationType = (FBType)destinationType.getTypeEntry().getType();
+		return true;
+	}
+
+	private boolean updateFBs() {
+		// Update all instances of edited FBTypes
+		updateCommands = new CompoundCommand();
+		getElementsOfType(sourceType).stream().forEach(fbnelem -> updateCommands.add(new UpdateFBTypeCommand(fbnelem)));
+		getElementsOfType(destinationType).stream()
+				.forEach(fbnelem -> updateCommands.add(new UpdateFBTypeCommand(fbnelem)));
+		updateCommands.execute();
+		return true;
+	}
+
+	private boolean createMUX() {
+		muxCreateCommand = new CompoundCommand();
+
+		structConnectionMap = new HashMap<>();
+		demuxMap = new HashMap<>();
+		muxMap = new HashMap<>();
+
+		// Create map between correct connected FBs and create MUX
+		getElementsOfType(destinationType).stream().forEach(instance -> {
+			instance.getInterface().getErrorMarker().stream().flatMap(err -> err.getInputConnections().stream());
+
+			// Collect all correct connections
+			final List<Connection> cons = instance.getInterface().getErrorMarker().stream()
+					.flatMap(err -> err.getInputConnections().stream())
+					.filter(con -> replacableConMap.containsKey(con.getSource().getName())
+							&& replacableConMap.get(con.getSource().getName()).equals(con.getDestination().getName())
+							&& con.getSourceElement().getType().getName().equals(sourceType.getName()))
+					.toList();
+
+			// Check if all connections between 2 instances are correct -> store in map
+			if (cons.stream()
+					.map((Function<? super Connection, ? extends FBNetworkElement>) Connection::getSourceElement)
+					.distinct().count() == 1 && cons.stream().count() == replacableConMap.size()) {
+				structConnectionMap.put(instance,
+						instance.getInterface().getErrorMarker().stream()
+								.flatMap(err -> err.getInputConnections().stream())
+								.filter(con -> replacableConMap.containsValue(con.getDestination().getName()))
+								.findFirst().get().getSourceElement());
+				// Create MUX if selected in wizard
+			} else if (conflictResolution) {
+				final FBCreateCommand muxcreate = new FBCreateCommand(
+						instance.getTypeLibrary().getFBTypeEntry("STRUCT_MUX"), instance.getFbNetwork(), 0, 0);// TODO:
+																												// (int)(instance.getPosition().getX()+10),
+																												// (int)instance.getPosition().getY());
+				muxCreateCommand.add(muxcreate);
+				muxcreate.execute();
+				final ChangeStructCommand changeStruct = new ChangeStructCommand(
+						(StructManipulator) muxcreate.getElement(), structType);
+				muxCreateCommand.add(changeStruct);
+				changeStruct.execute();
+				muxMap.put(instance, changeStruct.getNewElement());
+			}
+		});
+
+		// Create DEMUX if selected in wizard if any connections can not be replaced
+		// with Struct
+		if (conflictResolution) {
```

Comment:
Please make an own command for that:
ReplaceBrokenConnectionsWithMultiplexerCommand

So that we can use it inside this command and additionally we can also use it standalone.

This means the following:
we first perform the refacotoring, and afterwards we apply the ReplaceBrokenCon.. for repairing the broken connections